contracts/Overtime/ChainlinkResolver.sol

```diff
@@ -0,0 +1,200 @@
+        string[] _playerIds;
+    }
+
+    mapping(bytes32 => MarketResolveData) public requestIdToMarketResolveData;
+
+    uint public requestCounter;
+    mapping(uint => bytes32) public counterToRequestId;
+
+    constructor(
+        address _link,
+        address _oracle,
+        address _sportsAMM,
+        address _resultManager,
+        bytes32 _jobSpecId,
+        uint _paymentAmount
+    ) Ownable(msg.sender) {
+        setChainlinkToken(_link);
+        setChainlinkOracle(_oracle);
+        linkToken = IERC20(_link);
+        sportsAMM = ISportsAMMV2(_sportsAMM);
+        resultManager = ISportsAMMV2ResultManager(_resultManager);
+        jobSpecId = _jobSpecId;
+        paymentAmount = _paymentAmount;
+    }
+
+    function requestMarketResolving(
+        string[] calldata _gameIds,
+        string[] calldata _typeIds,
+        string[] calldata _playerIds
+    ) external whenNotPaused {
+        require(
+            _gameIds.length > 0 && _gameIds.length == _typeIds.length && _typeIds.length == _playerIds.length,
+            "Requested data has to be of same length"
+        );
+        Chainlink.Request memory req;
+
+        req = buildChainlinkRequest(jobSpecId, address(this), this.fulfillMarketResolve.selector);
+
+        req.addStringArray("_gameIds", _gameIds);
+        req.addStringArray("_typeIds", _typeIds);
+        req.addStringArray("_playerIds", _playerIds);
+
+        bytes32 requestId = sendChainlinkRequest(req, paymentAmount);
+
+        _putLink(msg.sender, paymentAmount);
+
+        MarketResolveData memory data = MarketResolveData(requestId, _gameIds, _typeIds, _playerIds);
+        requestIdToMarketResolveData[requestId] = data;
+
+        counterToRequestId[requestCounter++] = requestId;
+
+        emit MarketResolvingRequested(msg.sender, _gameIds, _typeIds, _playerIds);
+    }
+
+    function fulfillMarketResolve(
+        bytes32 _requestId,
+        int24[][] calldata _results
+    ) external whenNotPaused recordChainlinkFulfillment(_requestId) {
+        //might be redundant as already done by Chainlink Client, but making double sure
+        require(!requestIdFulfilled[_requestId], "Request ID already fulfilled");
+
+        MarketResolveData memory marketData = requestIdToMarketResolveData[_requestId];
+
+        bytes32[] memory _gameIds = new bytes32[](marketData._gameIds.length);
+        for (uint i = 0; i < marketData._gameIds.length; i++) {
+            _gameIds[i] = stringToBytes32(marketData._gameIds[i]);
+        }
+
+        uint16[] memory _typeIds = new uint16[](marketData._typeIds.length);
+        for (uint i = 0; i < marketData._typeIds.length; i++) {
+            _typeIds[i] = uint16(st2num(marketData._typeIds[i]));
+        }
+
+        uint16[] memory _playerIds = new uint16[](marketData._playerIds.length);
+        for (uint i = 0; i < marketData._playerIds.length; i++) {
+            _playerIds[i] = uint16(st2num(marketData._playerIds[i]));
+        }
```

Comment:
Since `_gameIds`, `_typeIds` and `_playerIds` are the same length, this can go under one loop.